### ISSUE TITLE
Create type safe way to pass workflow arguments to `ScheduleWorkflow`

### ISSUE THREAD

**Issue by bergundy** [sw-yx/temporal-time-utils#1]
Current example in README is both type unsafe and doesn't work because functions are not JSON serializable, this helper addresses the issue and adds type-safety:

```ts
Schedule.args = <W extends Workflow>(
  workflow: W | string, opts: ScheduleWorkflowOptions<W>
): [string, ScheduleWorkflowOptions<W>] {
    return [typeof workflow === 'string' ? workflow : workflow.name, args];
}

client.execute(Schedule, { args: Schedule.args(myWorkflow3, { args: [1, 2] }) })
```

Also this type uses the child workflow options which aren't the same as the client's workflow options, you might want to fix that while you're at it https://github.com/sw-yx/temporal-time-utils/blob/d2f6d7aebcc6c537c518e49abadd1245ce4b126d/packages/temporal-time-utils/ScheduleWorkflow.ts#L30

**Comment by swyxio** [comment]
it works because i just [grab the name of the function if given a function](https://github.com/sw-yx/temporal-time-utils/blob/d2f6d7aebcc6c537c518e49abadd1245ce4b126d/packages/temporal-time-utils/ScheduleWorkflow.ts#L53-L56) 🤷‍♂️ but appreciate the other points... will get to it eventually

**Comment by swyxio** [comment]
> Also this type uses the child workflow options which aren't the same as the client's workflow options

regarding this - i actually think this is correct, because these types are meant for the child workflow, NOT the top/parent/scheduler workflow

**Comment by bergundy** [comment]
> it works because i just [grab the name of the function if given a function](https://github.com/sw-yx/temporal-time-utils/blob/d2f6d7aebcc6c537c518e49abadd1245ce4b126d/packages/temporal-time-utils/ScheduleWorkflow.ts#L53-L56) 🤷‍♂️ but appreciate the other points... will get to it eventually

I don't think it should work because you need to serialize the function into a payload which isn't possible

**Comment by bergundy** [comment]
> > Also this type uses the child workflow options which aren't the same as the client's workflow options

You're correct, I forgot that the workflow is always executed as a child

**Comment by swyxio** [comment]
> I don't think it should work because you need to serialize the function into a payload which isn't possible

ok ok i understand you now. good point
